logstash-core/src/main/java/co/elastic/logstash/api/Configuration.java

```diff
@@ -0,0 +1,106 @@
+package co.elastic.logstash.api;
+
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * LS Configuration example. Should be implemented like Spark config or Hadoop job config classes.
+ */
+public final class Configuration {
+
+    private final Map<String, Object> rawSettings;
+
+    /**
+     * @param raw Configuration Settings Map. Values are serialized.
+     */
+    public Configuration(final Map<String, Object> raw) {
+        this.rawSettings = raw;
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T> T get(final PluginConfigSpec<T> configSpec) {
```

Comment:
I think we have a gap in validation that could lead to the silent acceptance of bad input that is then ignored.

Because this method is only ever called from the implementations with the `PluginConfigSpec`s that the particular implementation knows about, I _believe_ we have nothing to validate that plugins correctly report errors when a pipeline configuration attempts to use config directives that the implementation does not know about.

Example:

~~~
input {
  stdin {
    codek => typo
  }
} 
~~~

I do see that we have a `Plugin#configSchema` that must return a `Collection<PluginConfigSpec<?>>, but I do not see it used anywhere to perform this validation.

Reply:
Agreed. It wasn't explicit that config validation would be part of the beta phase of the API, so I just now added it as a requirement on the [Java API meta issue](https://github.com/elastic/logstash/issues/9215).